File: bench/type4/adversarial/cases/hof_oracle/hof_error.js
function mapError(xs) {
  return xs.map((x) => 1 / 0);
}

function mapErrorLoop(xs) {
  const out = [];
  for (const x of xs) {
    out.push(1 / 0);
  }
  return out;
}

function emptyFlatMap(xs) {
  return [].flatMap((x) => [1 / 0]);
}

function scalarFlatMap(xs) {
  return xs.flatMap((x) => x);
}

File: bench/type4/adversarial/cases/hof_oracle/hof_error.py
def map_error(xs):
    return [1 / 0 for x in xs]


def map_error_loop(xs):
    out = []
    for x in xs:
        out.append(1 / 0)
    return out


def empty_flat_map(xs):
    return [1 / 0 for x in [] for y in xs]
